server/test/profile.spec.js

```diff
@@ -162,7 +162,7 @@ describe('Get user data', () => {
           .end((err, res) => {
             expect(res.status).to.equal(404);
             expect(res.body.message).to.be.a('string');
-            expect(res.body.message).to.equal('user not found');
+            expect(res.body.message).to.equal('User with username john70 does not exist');
             done();
```

Comment:
Is there a reason why john70 was hardcoded here, kindly confirm

Reply:
I wrote a middleware to validate the username param in the route because I needed it for my feature and I made the error message more descriptive. 

I then updated the existing routes on the platform that were performing similar checks to avoid code duplication and I updated a test that was supposed to fail because a user with the username 'john70' does not exist to return the more descriptive error message instead of just 'user not found'